src/key.rs

```diff
@@ -78,6 +97,34 @@ impl ChallengeResponseKey {
                     crate::crypt::calculate_hmac_sha1(&[&challenge], &secret_bytes)?.to_vec();
                 Ok(response)
             }
+            ChallengeResponseKey::YubikeyChallenge(slot_number) => {
+                let mut yubikey_client = Yubico::new();
+                let slot = parse_yubikey_slot(slot_number)?;
+
+                let yubikey_device = match yubikey_client.find_yubikey() {
```

Comment:
This doesn't seem to find a specific yubikey, so if more then one is plugged in you'll get the first one in the USB chain.

Reply:
@droidmonkey that is correct, and I realize that we are probably doing the same thing in `keepassxc-cli`. I [opened a PR](https://github.com/wisespace-io/yubico-manager/pull/13) in the yubico-manager library to at least return all the yubikeys connected to the system. With that PR we can at least detect that multiple yubikeys are connected, and print a warning or return an error. Ideally we would be able to select the yubikey using the serial number, but that won't be as straightforward to add to the library.

Reply:
I already did that work, which is why we embedded the ykcore code into our code base. They wouldn't accept the PR I gave them before. In the cli we allow specifying yubikey by slot and optionally serial `slot[:serial]`

Reply:
I updated the PR to support multiple yubikeys. I added a `get_yubikey` function that takes an optional serial number. If the serial number is not provided, we assume that there is exactly one yubikey present on the system, and throw an error if that is not the case. Otherwise, we search for the yubikey with that serial number.

This PR is based on an [unmerged branch](https://github.com/wisespace-io/yubico-manager/pull/16) in the yubico-manager repository, so I'm going to convert this PR to a draft until the yubico-manager PR is merged.

Reply:
@droidmonkey The new feature is now using a stable release of [the `challenge-response` library](https://github.com/louib/challenge-response), so this PR is ready for another review.